File: src/game/save/characterName.ts
import { CHARACTER_NAME_MAX_LENGTH } from './types'

export function normalizeCharacterName(raw: string): string | null {
  const trimmed = raw.trim().replace(/\s+/g, ' ')
  if (!trimmed) return null
  return trimmed.slice(0, CHARACTER_NAME_MAX_LENGTH)
}

export function isValidCharacterName(raw: string): boolean {
  return normalizeCharacterName(raw) != null
}

File: packages/ik_rules/lib/src/save/character_name.dart
import 'generated/save_models.dart';

final RegExp _whitespaceRun = RegExp(r'\s+');

/// Trim, collapse internal runs of whitespace, and clip to the stored length.
String? normalizeCharacterName(String raw) {
  final trimmed = raw.trim().replaceAll(_whitespaceRun, ' ');
  if (trimmed.isEmpty) return null;
  return trimmed.length <= characterNameMaxLength
      ? trimmed
      : trimmed.substring(0, characterNameMaxLength);
}

bool isValidCharacterName(String raw) => normalizeCharacterName(raw) != null;
